argschema/argschema_parser.py

```diff
@@ -154,7 +155,7 @@ def __init__(self,
         self.logger.debug('input_data is {}'.format(input_data))
 
         # convert schema to argparse object
-        p = utils.schema_argparser(self.schema)
+        p = utils.schema_argparser(self.schema, parser=parser)
```

Comment:
what if we made a method on ArgSchemaParser of 

```
def get_parser(self):
   return utils.schema_argparser(self.schema)
```

and that way extending ArgSchemaParser would be cleaner.